contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,558 @@
+type CStorPoolInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec           CStorPoolInstanceSpec `json:"spec"`
+	Status         CStorPoolStatus       `json:"status"`
+	VersionDetails VersionDetails        `json:"versionDetails"`
+}
+
+// CStorPoolInstanceSpec is the spec listing fields for a CStorPoolInstance resource.
+type CStorPoolInstanceSpec struct {
+	// HostName is the name of kubernetes node where the pool
+	// should be created. This is filled by CSPC-operator.
+	HostName string `json:"hostName"`
+	// NodeSelector is the labels that is filled by CSPC-operator, only
+	// for informational purpose.
+	NodeSelector map[string]string `json:"nodeSelector"`
+	// PoolConfig is the default pool config that applies to the
+	// pool on node.
+	PoolConfig PoolConfig `json:"poolConfig"`
+	// RaidGroups is the group containing block devices
+	RaidGroups []RaidGroup `json:"raidGroup"`
+}
+
+// CStorPoolStatus is for handling status of pool.
+type CStorPoolStatus struct {
+	Phase    CStorPoolPhase        `json:"phase"`
+	Capacity CStorPoolCapacityAttr `json:"capacity"`
+	// LastTransitionTime refers to the time when the phase changes
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	LastUpdateTime     metav1.Time `json:"lastUpdateTime,omitempty"`
+	Message            string      `json:"message,omitempty"`
+}
+
+// CStorPoolCapacityAttr stores the pool capacity related attributes.
+type CStorPoolCapacityAttr struct {
+	Total string `json:"total"`
+	Free  string `json:"free"`
+	Used  string `json:"used"`
+}
+
+// CStorPoolInstanceList is a list of CStorPoolInstance resources
+type CStorPoolInstanceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorPoolInstance `json:"items"`
+}
+
+```
+
+Following improvments can be made to the schema :
+- The capacity fields ( i.e. Total, Free and Used) can be made to use Qunatity type that is helpful in capacity parsing and comparisons.
+- Status can be improved to include `Conditions` which can describe the state of pool in more detail.
+
+Following the the proposed schema for Status and Capacity (i.e. CStorPoolStatus and CStorPoolCapacityAttr struct ).
+(Also renaming th struct CStorPoolStatus to CStorPoolInstanceStatus and CStorPoolCapacityAttr to CStorPoolInstanceCapacity)
+
+```go
+
+// CStorPoolInstanceStatus is for handling status of pool.
+type CStorPoolInstanceStatus struct {
+	
+	// Current state of CSPI with details.
+	Conditions []CStorPoolInstanceCondition 
+	//  The phase of a CStorPool is a simple, high-level summary of the pool state on the 
+	//  node.  
+	Phase    CStorPoolPhase        `json:"phase"`
+	// Capacity describes the capacity details of a cstor pool 
+	Capacity CStorPoolCapacityAttr `json:"capacity"`
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
```

Comment:
Why we need `LastTransitionTime` and `LastUpdateTime` in `status` if its already present in `Conditions` ?

Reply:
Not required. Will remove this. 
( cc @mittachaitu -- Do you know why it was added at the first place?  )

Reply:
this is to capture change in Phase which is different from the ones in condition

Reply:
As discussed the transition field has been removed.